src/meta/metainfo.go

```diff
@@ -492,16 +492,14 @@ func NameEquals(n *name.Name, s string) bool {
 		return false
 	}
 
-	sParts := strings.Split(s, `\`)
-
+	rest := s
 	for i, part := range n.Parts {
-		p, ok := part.(*name.NamePart)
-		if !ok {
+		part := part.(*name.NamePart)
+		if !strings.HasPrefix(rest, part.Value) {
```

Comment:
This code is slightly incorrect. I think we also need to check two things:

1. That there is a character `\` after `part.Value` ends if it's not the final part (essentially that the prefix is `part.Value + "\\"`).
2.  If it is the final part we need to check that there are no more characters in the string.

Otherwise the following would be equal:

`a\b` = `a\bb` because we only check the prefixes
`a b\` = `a\b` because we don't check where is `\` symbol

Reply:
Good catch, thanks!

I believe if we handle the last part separately with "==" instead of "hasPrefix", all issues go away.

I've added test cases that check whether the mentioned scenarios lead to expected results. 

> 1 `a\b` = `a\bb` because we only check the prefixes

Now we check the last part with `==`, should be fixed

> 2. `a b\` = `a\b` because we don't check where is \ symbol

This will fail in old code as well, since the second part will be checked as
`HasPrefix("b", "")`, which is false. In updated code it shouldn't be an issue as well, as `"b"!=""`.

> 1. That there is a character \ after part.Value ends if it's not the final part (essentially that the prefix is part.Value + "\\").

I can't make up an example that will pass NameEquals with something except `\` as a separator. It will cause either different numbers of parts (we have an early check for it) or difference during the next part comparison. If you have an example, I can add it to tests and then it will be easier to fix it.